include/crm/lrmd.h

```diff
@@ -289,115 +299,171 @@ typedef struct lrmd_api_operations_s {
+     * \param[in]     client_name  Arbitrary identifier to pass to server
+     * \param[out]    fd           If not NULL, where to store file descriptor
+     *                             for connection's socket
+     *
      * \return Legacy Pacemaker return code
      */
-    int (*connect) (lrmd_t * lrmd, const char *client_name, int *fd);
+    int (*connect) (lrmd_t *lrmd, const char *client_name, int *fd);
 
     /*!
      * \brief Initiate an executor connection without blocking
      *
+     * \param[in,out] lrmd         Executor connection object
+     * \param[in]     client_name  Arbitrary identifier to pass to server
+     * \param[in]     timeout      Error if not connected within this time
+     *                             (milliseconds)
+     *
      * \return Legacy Pacemaker return code (if pcmk_ok, the event callback will
      *         be called later with the result)
      * \note This function requires a mainloop.
      */
-    int (*connect_async) (lrmd_t * lrmd, const char *client_name, int timeout /*ms */ );
+    int (*connect_async) (lrmd_t *lrmd, const char *client_name,
+                          int timeout /*ms */ );
 
     /*!
      * \brief Check whether connection to executor daemon is (still) active
      *
+     * \param[in,out] lrmd  Executor connection object to check
+     *
      * \return 1 if the executor connection is active, 0 otherwise
      */
-    int (*is_connected) (lrmd_t * lrmd);
+    int (*is_connected) (lrmd_t *lrmd);
 
     /*!
-     * \brief Poke executor connection to verify it is still capable of serving requests
-     * \note The response comes in the form of a poke event to the callback. 
+     * \brief Poke executor connection to verify it is still active
+     *
+     * \param[in,out] lrmd  Executor connection object to check
      *
      * \return Legacy Pacemaker return code (if pcmk_ok, the event callback will
      *         be called later with the result)
+     * \note The response comes in the form of a poke event to the callback.
+     *
      */
-    int (*poke_connection) (lrmd_t * lrmd);
+    int (*poke_connection) (lrmd_t *lrmd);
 
     /*!
      * \brief Disconnect from the executor.
      *
+     * \param[in,out] lrmd  Executor connection object to disconnect
+     *
      * \return Legacy Pacemaker return code
      */
-    int (*disconnect) (lrmd_t * lrmd);
+    int (*disconnect) (lrmd_t *lrmd);
 
     /*!
-     * \brief Register a resource with the executor.
+     * \brief Register a resource with the executor
+     *
+     * \param[in,out] lrmd      Executor connection object
+     * \param[in]     rsc_id    ID of resource to register
+     * \param[in]     standard  Resource's resource agent standard
+     * \param[in]     provider  Resource's resource agent provider (or NULL)
+     * \param[in]     agent     Resource's resource agent name
+     * \param[in]     options   Group of enum lrmd_call_options flags
```

Comment:
For the options marked as `Group of enum lrmd_call_options flags`, is that the case? The function argument is an enum value, not a `uintX_t` value (though a quick test indicates there is no checking, probably due to int promotion). More importantly, the `lrmd_call_options` enum is sequential integers, not sequential powers of 2. So I don't think it makes sense to use multiple of them together in general.

Reply:
They are intended to be flags :-/ but thankfully only one is ever set by current callers. However that does mean the checks for lrmd_opt_drop_recurring and lrmd_opt_notify_orig_only in execd_commands.c will currently also hit if the other one is set. :-/

I think we should stick with "Group of" here since that's the intent. The argument should be uint32_t but it's public API so we can't change it (there are quite a few old functions that advertise enums where a mask is actually used). I can comment the situation and add workarounds for the broken checks until we can refactor.